### PR TITLE
Migrate to App router

### LINKED ISSUE #42: Migrate to app router

### PR DESCRIPTION
**Pull request by isofore**
Closes #42 

- Move from Pages router to App router
- Managed to remove some APIs
- Tried to use RSC where possible, data fetching got a lot nicer

For the future:
- Revalidation has to be checked (not sure if the changes I made work) #45 
- We could move away from `marked` to something like https://github.com/hashicorp/next-mdx-remote, which is one of the libraries recommended by Next.js

### CODE REVIEW

File: app/medlem/[privatePage]/layout.tsx [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import React from 'react'
+
+const MedlemPageLayout = ({ children }: { children: React.ReactNode }) => (
+  <div className="min-h-screen">
```
Comment (backjonas) [inline app/medlem/[privatePage]/layout.tsx:4]: This layout is duplicated for medlem page, content page & post(s), could be extracted to some PageLayout component

File: app/api/revalidate/route.ts [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    crypto.timingSafeEqual(Buffer.from(bearer), Buffer.from(API_KEY))
+  ) {
+    return NextResponse.json({ error: 'Invalid token' }, { status: 401 })
+  }
+
+  try {
+    const tag = request.nextUrl.searchParams.get('tag')
+    if (tag) {
+      // Manual API call
+      revalidateTag(tag)
```
Comment (backjonas) [inline app/api/revalidate/route.ts:29]: For revalidateTag to work the tags have to be provided in the fetch request, so currently this doesn't do anything.

File: app/page.tsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
+// import Image from 'next/image'
```
Comment (backjonas) [inline app/page.tsx:1]: Remove

File: package.json [hunk 4]
```diff
@@ -6,10 +6,10 @@
   },
   "private": true,
   "scripts": {
-    "dev": "NODE_OPTIONS='--dns-result-order=ipv4first' next dev --turbo",
+    "dev": "NODE_OPTIONS='--dns-result-order=ipv4first' next dev",
```
Comment (backjonas) [inline package.json:9]: What's the reason for removing turbopack? No longer supported?

**Review by backjonas — APPROVED** [review]
Good work! Started to look a bit at refactoring the revalidate logic already so I can continue with that in a separate PR

It throws _Module not found: Can’t resolve ‘encoding’_ when I visit a page locally so that package should probably be added

File: app/api/revalidate/route.ts (hunk 2 above)
Comment (isofore) [reply app/api/revalidate/route.ts:29]: Yeah, my thought was that this will be properly done in #45

**Pull request merged**